lib/money/money.rb

```diff
@@ -74,6 +75,16 @@ def from_subunits(subunits, currency_iso, format: :iso4217)
       new(value, currency)
     end
 
+  def from_json(hash_or_string)
```

Comment:
I wonder if this might be a bit easier to follow with explicit, separate methods `from_hash` and `from_json` that each handle a single data type, rather than attempting to juggle types within a single method. It seems a slightly surprising result to have a method from_json that accepts an input which is not JSON.

Reply:
Do we need  `from_hash` at all? I propose only starting with `from_json`. Going from a hash to a Money object is trivial `Money.new(hash[:amount], hash[:currency])`